src/malva/java/text/DecimalFormatTest.java

```diff
@@ -19,7 +23,49 @@ public static void testFormat() {
     assertEquals("-4,3 unit", df.format(-4.3));
   }
 
+  public static void testFormatToCharacterIterator() {
+    final DecimalFormat df = new DecimalFormat("0.##");
+
+    // Check first for throwables
+    assertThrows(new Block() {
+      @Override public void run() {
+        df.formatToCharacterIterator(null);
+      }
+    }, NullPointerException.class);
+
+    assertThrows(new Block() {
+      @Override public void run() {
+        df.formatToCharacterIterator(new Object());
+      }
+    }, IllegalArgumentException.class);
+
+    // Attributes check
+    Set<?> attrs;
+    attrs = formatAndGetAttrs(df, 0);
+    assertTrue(attrs.containsAll(Arrays.asList(Field.INTEGER)));
+
+    attrs = formatAndGetAttrs(df, Double.valueOf(1.23));
+    assertTrue(attrs.containsAll(Arrays.asList(Field.INTEGER, Field.DECIMAL_SEPARATOR, Field.FRACTION)));
```

Comment:
This checks that all the specified attributes are included in the resulting set, but shouldn't we also make sure that there are no additional attributes other than the ones that we expect ?

Reply:
For example you could create a helper method that checks with `attrs.containsAll`, but then additionally checks that the size of the set equals the number of expected elements.

Reply:
Yes, I thought about this.. I'll fix it :+1: .